File: Backend/Models/Post.cs
using System.Text.Json.Serialization;

namespace Backend.Models;

public class Post
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string PostImage { get; set; } = string.Empty;

    [JsonConverter(typeof(PostCategoryJsonConverter))]
    public PostCategory Category { get; set; }

    public Location? Location { get; set; }
    public DateTime DatePosted { get; set; }
    public User UserPosted { get; set; } = new();
}

File: frontend/src/types/post.ts
export type PostCategory = 'buy&sale' | 'events'

export interface Location {
  latitude: number
  longitude: number
}

export interface User {
  id: string
  name: string
  avatar: string
}

export interface Post {
  id: string
  title: string
  description: string
  postImage: string
  category: PostCategory
  location: Location | null
  datePosted: string
  userPosted: User
}

export interface PagedPosts {
  pageNumber: number
  itemCount: number
  totalCount: number
  totalPages: number
  items: Post[]
}

export interface PostDto {
  title: string
  description: string
  postImage: string
  category: PostCategory
  location: Location | null
  userPosted: User
}

export interface PostFilters {
  q: string
  category: PostCategory | ''
  dateFrom: string
  dateTo: string
}
